core/blockchain.go

```diff
@@ -221,20 +220,22 @@ type BlockChain struct {
 	setPrivateState func([]*types.Log, *state.StateDB, types.PrivateStateIdentifier) // Function to check extension and set private state
 
 	// Quorum
-	isMultitenant bool // if this blockchain supports multitenancy
+	quorumConfig QuorumChainConfig // quorum chain config holds all the possible configuration fields for GoQuorum
 	// privateStateManager manages private state(s) for this blockchain
 	privateStateManager mps.PrivateStateManager
-	saveRevertReason    bool // if we should save the revert reasons in the Tx Receipts
 	// End Quorum
 }
 
 // NewBlockChain returns a fully initialised block chain using information
 // available in the database. It initialises the default Ethereum Validator and
 // Processor.
-func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *params.ChainConfig, engine consensus.Engine, vmConfig vm.Config, shouldPreserve func(block *types.Block) bool, txLookupLimit *uint64) (*BlockChain, error) {
+func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *params.ChainConfig, engine consensus.Engine, vmConfig vm.Config, shouldPreserve func(block *types.Block) bool, txLookupLimit *uint64, quorumChainConfig *QuorumChainConfig) (*BlockChain, error) {
```

Comment:
Hope there's a way not to change the original method signature so to ease the upstream merge effort.

Reply:
I did a variadic variable but got reviewed earlier

Reply:
the goal was to in the future only have this config struct for any necessary quorum config, requiring less changes in future upstream merges.

Reply:
but without making it unclear with the variadic variable. @trung what do you think?

Reply:
LGTM

Reply:
keep as it is or changing to variadic variable?